pkg/packetbrokeragent/grpc_gspba.go

```diff
@@ -211,3 +248,41 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 	}
 	return res, nil
 }
+
+func (s *gsPbaServer) fetchUserFromContact(
+	ctx context.Context, contact *ttnpb.OrganizationOrUserIdentifiers,
+	selectOrgContact func(*ttnpb.Organization) *ttnpb.OrganizationOrUserIdentifiers,
+) (*ttnpb.User, error) {
+	usrID := contact.GetUserIds()
+
+	// If the contact is an organization, get its contact before fetching the user information.
+	if orgID := contact.GetOrganizationIds(); orgID != nil {
+		org, err := s.entityRegistry.GetOrganization(ctx, &ttnpb.GetOrganizationRequest{
+			OrganizationIds: orgID,
+			FieldMask:       ttnpb.FieldMask("administrative_contact", "technical_contact"),
+		})
+		if err != nil {
+			return nil, err
+		}
+		usrID = selectOrgContact(org).GetUserIds()
+	}
+
+	usr, err := s.entityRegistry.GetUser(ctx, &ttnpb.GetUserRequest{
+		UserIds:   usrID,
+		FieldMask: ttnpb.FieldMask("ids", "primary_email_address", "primary_email_address_validated_at"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if usr.PrimaryEmailAddressValidatedAt == nil {
```

Comment:
We could force this check to be done in the caller code in order to avoid skipping all of the failed precondition error codes.